### PR TITLE
(fleet) improve the installer install script

### PR DESCRIPTION
**Pull request by arbll**
- add the new boostrap call
- remove the on-disk installer file
- add support for RPM
- setting DD_INSTALLER will always install, regardless of APM being enabled
- add `DD_REMOTE_UPDATES`

### CODE REVIEW

**Review by BaptisteFoy — APPROVED** [review]

File: install_script.sh.template [hunk 1]
```diff
@@ -498,53 +498,49 @@ END
...
+    local datadog_installer="$4"
+    local apm_instrumentation_enabled="$5"
 
-    if [ "$os" == "Debian" ] && [ "$distribution" == "Debian" ] && [ -n "$apm_instrumentation_enabled" ]; then
+    if [ "$os" == "Debian" ]; then
         echo -e "\033[34m\n* Installing the Datadog installer\n\033[0m"
-        configure_installer "$sudo_cmd" "$trace_id" "$apm_instrumentation_enabled"
-        $sudo_cmd apt-get install -o Acquire::Retries="5" -y --force-yes "datadog-installer"
+        $sudo_cmd apt-get install -o Acquire::Retries="5" -y --force-yes "datadog-installer" || return $?
+    elif [ "$os" == "RedHat" ]; then
```
Comment (raphaelgavache) [inline install_script.sh.template:511]: you should enable the test suite for yum too

File: install_script.sh.template [hunk 2]
```diff
@@ -1631,6 +1645,17 @@ elif [ ! "$no_agent" ]; then
     update_hostname "$sudo_cmd" "$hostname" "$config_file"
     update_hosttags "$sudo_cmd" "$host_tags" "$config_file"
     update_env "$sudo_cmd" "$dd_env" "$config_file"
+    if [ -n "$remote_updates" ]; then
```
Comment (raphaelgavache) [inline install_script.sh.template:1648]: nit: moving this to its own function / in update_remote_updates would make it easier to read

File: install_script.sh.template [hunk 3]
```diff
@@ -498,53 +498,49 @@ END
...
-EOT"
-}
-
 function _install_installer() {
     local sudo_cmd="$1"
     local os="$2"
     local distribution="$3"
-    local apm_instrumentation_enabled="$4"
+    local datadog_installer="$4"
+    local apm_instrumentation_enabled="$5"
```
Comment (raphaelgavache) [inline install_script.sh.template:506]: missed the cleanups of datadog_installer and apm_instrumentation_enabled which are now unused: check moved above

**Review by raphaelgavache — APPROVED** [review]
lgtm, left some minor comments

**Review by raphaelgavache — APPROVED** [review]

**Review by Pythyu — APPROVED** [review]
Looks good 🚀

**Pull request merged**